File: frontend-admin/src/api/circle.ts
import { get, post, put } from './request'
import type { PageQuery, PageResult } from './types'

/**
 * 圈子管理 API
 * 功能：创建申请审核、编辑、下架、转让圈主、分类、推荐位
 */

/** 圈子分类 */
export interface CircleCategory {
  id: number
  name: string
  sort: number
  status: number
}

/** 圈子信息 */
export interface CircleInfo {
  circleId: number
  ownerId: number
  ownerName: string
  name: string
  categoryId: number
  categoryName: string
  intro: string
  type: 'PUBLIC' | 'PRIVATE'
  memberCount: number
  postCount: number
  status: string
  recommended: boolean
  icon?: string
  createTime: string
}

/** 圈子创建申请 */
export interface CircleCreationApply {
  applyId: number
  ownerId: number
  ownerName: string
  name: string
  categoryId: number
  categoryName: string
  intro: string
  avatar: string
  type: 'PUBLIC' | 'PRIVATE'
  applyTime: string
  status: 'PENDING' | 'REJECTED'
}

/**
 * 查询圈子列表
 */
export function getCircleList(params: PageQuery & { status?: string | number; categoryId?: number }) {
  return get<PageResult<CircleInfo>>('/circle/list', params)
}

/**
 * 查询圈子创建申请列表
 */
export function getCircleCreationApplyList(params: PageQuery) {
  return get<PageResult<CircleCreationApply>>('/circle/creation-apply-list', params)
}

/**
 * 审批通过圈子创建
 */
export function approveCircleCreation(circleId: number) {
  return post('/circle/creation-approve', { circleId })
}

/**
 * 驳回圈子创建
 */
export function rejectCircleCreation(circleId: number, reason: string) {
  return post('/circle/creation-reject', { circleId, reason })
}

/**
 * 编辑圈子信息
 */
export function updateCircle(data: Partial<CircleInfo> & { circleId: number }) {
  return put('/circle/update', data)
}

/**
 * 下架圈子
 */
export function offlineCircle(circleId: number, reason: string) {
  return post('/circle/offline', { circleId, reason })
}

/**
 * 转让圈主
 */
export function transferOwner(circleId: number, newOwnerId: number) {
  return post('/circle/transfer', { circleId, newOwnerId })
}

/**
 * 设置推荐位
 */
export function setRecommend(circleId: number, recommended: boolean) {
  return post('/circle/recommend', { circleId, recommended })
}

/**
 * 获取圈子分类列表
 */
export function getCircleCategories() {
  return get<CircleCategory[]>('/circle/categories')
}

File: backend/src/main/java/com/minthive/entity/Circle.java
package com.minthive.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 兴趣圈子实体类
 *
 * <p>对应数据库 circle 表</p>
 */
@Data
@TableName("circle")
public class Circle implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 圈主用户ID */
    private Long ownerId;

    /** 圈子名称 */
    private String name;

    /** 分类ID(关联circle_category) */
    private Long categoryId;

    /** 分类名称(非数据库字段，查询时填充) */
    @TableField(exist = false)
    private String categoryName;

    /** 圈子简介 */
    private String intro;

    /** 圈子头像 */
    private String avatar;

    /** 圈子封面图 */
    private String banner;

    /** 圈子类型:0公开 1私密 */
    private Integer type;

    /** 成员数量 */
    private Integer memberCount;

    /** 圈子状态:0下架 1正常 */
    private Integer status;

    /** 是否推荐:0否 1是 */
    private Integer recommend;

    /** 圈内公告 */
    private String notice;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 逻辑删除 */
    @TableLogic
    @TableField("deleted")
    private Integer deleted;

    /** 当前用户是否已加入（非持久化字段，由后端查询 circle_member 表后填充） */
    @TableField(exist = false)
    private Boolean joined;

    /** 帖子数量（非持久化字段，由后端动态统计 post 表填充） */
    @TableField(exist = false)
    private Integer postCount;
}
